mm2src/coins/eth/v2_activation.rs

```diff
@@ -1013,10 +1015,11 @@ pub(super) async fn eth_shared_db_id(coin: &EthCoin, ctx: &MmArc) -> Option<Stri
 
 pub(super) async fn eth_account_db_id(coin: &EthCoin) -> Option<String> {
     match coin.derivation_method() {
-        DerivationMethod::HDWallet(hd_wallet) => hd_wallet
-            .get_enabled_address()
-            .await
-            .map(|addr| dhash160(addr.pubkey().as_bytes()).to_string()),
+        DerivationMethod::HDWallet(hd_wallet) => hd_wallet.get_enabled_address().await.map(|addr| {
+            // Skip the first byte of the uncompressed public key before converting to the eth address.
+            let pubkey = Public::from_slice(&addr.pubkey().as_bytes()[1..]);
+            public_to_address(&pubkey).to_string()
```

Comment:
I think you should use `display_eth_address` function and strip the `0x` at the start

Reply:
> I think you should use display_eth_address function and strip the 0x at the start

I think the below would be much simpler than what you implemented
```rust
let address = display_eth_address(&addr.address());
address.trim_start_matches("0x").to_string()
```